src/tablestore.rs

```diff
@@ -222,28 +229,45 @@ impl TableStore {
         handle: &SSTableHandle,
         blocks: Range<usize>,
     ) -> Result<VecDeque<Block>, SlateDBError> {
-        let path = self.path(&handle.id);
-        let obj = ReadOnlyObject {
-            object_store: self.object_store.clone(),
-            path,
-        };
-        self.sst_format
-            .read_blocks(&handle.info, blocks, &obj)
-            .await
+        let futures: Vec<_> = blocks
```

Comment:
I'm a little worried about this. If we don't have any of these blocks in the cache, then the `.read_block` call will trigger a `GET` to S3 for each block. This could cause S3 API bills to get very expensive. When we do a `read_blocks` right now, we end up using a `get_range` call which executes a single GET.

I'm not sure what the best approach is here.

1. We could ignore the cache for range reads, but range reads are pretty much all we do right now (see the dead_code macro for `read_block` below).
2. We could break the range up into sub-ranges for blocks we don't have in cache, and stitch them back together with the cached blocks we have.
3. We could cache block ranges. Not quite sure how well this would work.
4. Something else?